File: src/template-vb/Controllers/HomeController.vb
Imports Castle.MonoRail.Framework

Namespace Controllers
    <Layout("Default"), Rescue("500")> _
    Public Class HomeController
        Inherits SmartDispatcherController

        Sub Index()
            PropertyBag("message") = "Welcome to Castle MonoRail!"
        End Sub

        Sub About()
        End Sub
    End Class
End Namespace

File: src/template-cs/MonoRail.Template/MonoRail.Template/Controllers/HomeController.cs
using System;
using Castle.MonoRail.Framework;

namespace MonoRail.Template.Controllers
{
    [Layout("Default"), Rescue("500")]
    public class HomeController : SmartDispatcherController
    {
        public void Index()
        {
            PropertyBag["message"] = "Welcome to Castle MonoRail!";
        }

        public void About()
        {
        }
    }
}